### DISCUSSION TITLE
Doesn't work custom validation in input after updated from 5.7.2 to 6.8.2

### DISCUSSION THREAD

**Discussion by mateuszmaczynski** [react-hook-form/react-hook-form#2950] [Q&A] [answered]
Hi, I have a request, what I should do that it works, in version 5.7.2 it worked after updated to new vesion unfortunately doesn't work. My code below:

`<Controller
  as={Input}
  name="numberPesel"
  control={control}
  rules={{required: true}}
  onBlur={(e) => {
    const isValidPesel = validatePesel(e[0].target.value);
    if (!isValidPesel) {
      setError([{
        type: "invalidNumber",
        name: "numberPesel"
      }]);
      setPeselError(true);
    } else {
      setPeselError(false);
    }
  }}
/>`

**Comment by bluebill1049** [comment]
change to `render`?

**Reply by bluebill1049** [reply]
take a look at the changelog: https://github.com/react-hook-form/react-hook-form/blob/master/CHANGELOG.md

**Comment by mateuszmaczynski** [comment, marked as answer]
Thank you very much, now it works validation :)  My code looks like:

```<Controller
  control={control}
  name="numberPesel"
  rules={{required: true}}
  render={({onChange, onBlur, value}) => (
    <Input
      value={value}
      onChange={e => onChange(e)}
      onBlur={(e) => {
        const isValidPesel = validatePesel(e.target.value);
        if (!isValidPesel) {
          setError('numberPesel', {type: "invalidNumber", message: 'Wprowadzony numer pesel jest błędny"', shouldFocus: true });
          setPeselError(true);
        } else {
          setPeselError(false);
        }
      }}
      defaultValue={(userData !== null && userData.numberPesel) ? userData.numberPesel : ''}
    />
  )}```

**Reply by bluebill1049** [reply]
nice awesome to hear.
